systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/InternalKafkaClient.java

```diff
@@ -231,8 +231,12 @@ public void produceTlsMessagesUntilOperationIsSuccessful(int receivedMessages) {
     }
 
     public void consumesTlsMessagesUntilOperationIsSuccessful(int sentMessages) {
+        consumesTlsMessagesUntilOperationIsSuccessful(sentMessages, Constants.GLOBAL_TIMEOUT);
+    }
+
+    public void consumesTlsMessagesUntilOperationIsSuccessful(int sentMessages, long timeout) {
         TestUtils.waitFor("Consumer will successfully receive messages.", Duration.ofMinutes(1).toMillis(),
-            Constants.GLOBAL_TIMEOUT, () -> {
+            timeout, () -> {
```

Comment:
why this is needed?

Reply:
I made it more configurable (same method name, added timeout as param) for timeouts, as default was "only 5mins". In KMM2, it might take longer than 5mins). I Use 10mins iirc.

Reply:
You should maybe think if we can shorten the timeout -> instead of making it configurable to support 10 minutes timeout, can we make the timeout shorter?

Reply:
Exactly, even 5 minutes is quite long. I would expect that messages will be mirrored much quicker

Reply:
Yes, this method supports it of course :-) You can set there 1min as well...

Reply:
What I meant is to use it and not to support it. Why does it need to wait for 10 minutes here?

Reply:
Well, not exactly 10mins, but 5 mins are not enough sometimes.. 
```
2022-01-03 14:58:35 [main] INFO  [InternalKafkaClient:193] Consumer finished correctly: false
2022-01-03 14:58:35 [main] INFO  [InternalKafkaClient:196] Consumer consumed 0 messages
io.strimzi.test.WaitException: Timeout after 300000 ms waiting for Consumer will successfully receive messages.
	at io.strimzi.test.TestUtils.waitFor(TestUtils.java:158)
	at io.strimzi.test.TestUtils.waitFor(TestUtils.java:120)
	at io.strimzi.systemtest.kafkaclients.internalClients.InternalKafkaClient.consumesTlsMessagesUntilOperationIsSuccessful(InternalKafkaClient.java:238)
	at io.strimzi.systemtest.mirrormaker.MirrorMaker2ST.testKMM2RollAfterSecretsCertsUpdateScramsha(MirrorMaker2ST.java:12
	```

Reply:
Yeah, but why? What is it waiting for? What is happening during these 10 minutes? 10 minutes is quite long for just a random timeout - if there is something happening in the background, maybe it should be broken into smaller steps. But in any case we should know what are we waiting for and why we expect it to happen in 10 minutes and not in 5 or in 15.

Reply:
Well, it's KMM2 and all dependencies rolled after SHA password change. I just got one idea.. I will have a look at it. 
Seems like waiting only for KMM2 for RU (5mins before) is not enough. Maybe I should check for other components as well (which should be rolled by now as well, because KMM2 is one of the last to roll after password update, right?)

Reply:
Well, depends what password update are you talking about. But the operator treats Mirror Maker (1 & 2), Bridge or Connect independently. So it should not wait for anything before rolling it. 

Also, in general, if you know about the smaller steps in between, you should split the wait => e.g. instead of waiting 10 minutes for consumer to receive some message, you can wait for 2 minutes for password to be updated, 2 minutes for component to roll, etc. You are able to detect the failures faster and you get much better information about what actually failed

Reply:
Finally found the issue. I was setting value ```.addToConfig("refresh.topics.interval.seconds", 1)``` on KMM2 cluster spec instead of SourceConnectorSpec on KMM2Template. 
Now it finally works quickly without any timeouts.
Marking as resolved now.